src/openapi/transport/retry.js

```diff
@@ -116,8 +123,10 @@ TransportRetry.prototype.sendTransportCall = function(transportCall) {
         .apply(this.transport, transportCall.args)
         .then(transportCall.resolve,
             (response) => {
-                if (!(response && response.status) &&
-                    (transportCall.retryCount < this.methods[transportCall.method].retryLimit)) {
+                const callOptions = this.methods[transportCall.method];
+                if ((!(response && response.status) || callOptions.statuses && callOptions.statuses.indexOf(response.status) >= 0) &&

```

Comment:
perhaps it would be worth splitting out this if and assigning to variables first.. because there is now a lot of conditions, so it is difficult to follow.